ts/components/select/index.ts

```diff
@@ -258,32 +258,36 @@ export default (initOptions: InitOptions): Select => ({
     }
   },
   syncJsonOptions () {
-    this.setOptions(window.Alpine.evaluate(this, this.$refs.json.innerText))
+    setTimeout(() => {
+      this.setOptions(window.Alpine.evaluate(this, this.$refs.json.innerText))
 
-    if (this.hasWireModel) {
-      this.syncSelectedFromWireModel()
-    }
+      if (this.hasWireModel) {
+        this.syncSelectedFromWireModel()
+      }
+    })
   },
   syncSlotOptions () {
-    const elements = this.$refs.slot.querySelectorAll('[name="wireui.select.option"]')
-
-    const options = Array.from(elements).flatMap(element => {
-      const base64 = element.querySelector('[name="wireui.select.option.data"]')?.textContent
-
-      if (!base64) return []
-
-      const option: Option = window.Alpine.evaluate(this, base64)
-
-      option.html = element.querySelector('[name="wireui.select.slot"]')?.innerHTML
-
-      return option
+    setTimeout(() => {
```

Comment:
Have you tried using `$nextTick` or `requestAnimationFrame` instead a `setTimeout`?

Reply:
That's a good thought — I'll test an implementation in my working version and update here. I continued to experience issues in other contexts even after applying this change and traced it back to a problem in the `$wire.entangle()` implementation in Livewire's Alpine magic method.

While the order (with respect to MorphDOM's manipulations) still remained relevant, the bigger problem was in Alpine's enumeration of "dot notation" syntax for Livewire properties in contexts where Livewire may add/remove an `x-data`-assigned element, with a `$wire.entangle()` property, based on conditional Blade directives. I wound up adding a modified version of the `$wire` magic to fix this:

Here, I verify that the property is in-fact entangled via `x-data` in markup

https://gist.github.com/stephancasas/4e37883edfb230fbb5cc8570a4b91007#file-rewire-js-L42-L48

Then, here, I verify that the markup still exists prior to enumerating property values:

https://gist.github.com/stephancasas/4e37883edfb230fbb5cc8570a4b91007#file-rewire-js-L74-L81

---

My use case was a flyout-style component where I didn't want markup delivered to the DOM (even hidden) until the flyout was actually needed. I then wanted the ability to unmount the flyout's content post-dismissal. In practice, this meant enclosing Livewire-enabled components in a Blade `@if()`directive, which Livewire would then act on accordingly.

However, after the first flyout dismissal, the `<x-select>` component would fail entirely — along with any other `$wire.entangle()`-powered components inside of the `@if()` directive.

In truth, I'm not happy with using regex to solve the problem, but because I'm not certain that this is an issue worth raising in Alpine/Livewire (especially with v3 on the horizon) it works for now. Further, I realize that the problem could also be solved by swapping `@if()` for another display/render approach and/or deeper property initialization but, as I was building a component, I wanted the dev experience to feel rapid and a little more seamless as opposed to adding extra consideration.

What do you think?

Reply:
Hey, so sorry for my delay. I think in the next livewire version it will be easier to work on. Livewire and Alpine will be better integrated.
For now, we can use this workaround to fix it.
Is the setTimeout solution better than using **$nextTick** or **requestAnimationFrame**?

Reply:
Your [rewire.js](https://gist.github.com/stephancasas/4e37883edfb230fbb5cc8570a4b91007#file-rewire-js) is interesting
Do you think it will fix the select problem?